src/Node.js

```diff
@@ -0,0 +1,133 @@
+  // ### DOM Node Shim
+  //
+  // This represents a superficial form of a DOM node which contains most of
+  // the properties that a regular DOM node would contain, but only the ones
+  // needed by slowparse.
+module.exports = (function(){
+  "use strict";
+
+  function Node(nodeValue, namespaceURI) {
+    var self = this;
+    self.nodeValue = nodeValue || "";
+    self.namespaceURI = namespaceURI || "";
+    self.parentNode = false;
+    self.nextSibling = false;
+    self.isVoid = false;
+    self.childNodes = [];
+    self.attributes = [];
+    Object.defineProperties(self, {
+      "innerHTML": {
+        get: function() {
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return null;
+          }
+
+          var innerHTML = "";
+
+          self.childNodes.forEach(function(childNode) {
+            innerHTML += childNode.outerHTML || "";
+          });
+
+          return innerHTML;
+        }
+      },
+      "outerHTML": {
+        get: function() {
+          if(self.nodeType === self.ATTRIBUTE_NODE) {
+            return null;
+          }
+          if(self.nodeType === self.COMMENT_NODE) {
+            return "<!--" + self.nodeValue + "-->";
+          }
+          if(self.nodeType !== self.ELEMENT_NODE) {
+            return self.nodeValue;
+          }
+
+          var attributeStr = "";
+          self.attributes.forEach(function(attr) {
+            attributeStr += attr.nodeName + "=\"" + (attr.nodeValue || "") + "\" ";
+          });
+          attributeStr = attributeStr.trim();
+          if(attributeStr.length > 0) {
+            attributeStr = " " + attributeStr;
+          }
+
+          var openingTag = "<" + self.nodeName.toLowerCase() + attributeStr + ">";
+          var closingTag = self.isVoid ? "" : "</" + self.nodeName.toLowerCase() + ">";
+
+          return openingTag + (self.innerHTML || "") + closingTag;
+        }
+      }
+    });
+  }
+
+  // Add a node to the current node as a child
+  Node.prototype.appendChild = function(node) {
```

Comment:
Nit, but don't split up the prototype into a million Thing.prototype.fname = ... calls. Just use
```
Node.prototype = {
  fname1: function(...) {
    ...
  },
  fname2: ...
  ...
}
```